File: src/main/java/io/lsdconsulting/lsd/distributed/generator/diagram/event/MessageBuilder.java
package io.lsdconsulting.lsd.distributed.generator.diagram.event;

import com.lsd.core.IdGenerator;
import com.lsd.core.domain.ComponentName;
import com.lsd.core.domain.Message;
import com.lsd.core.domain.MessageType;
import com.lsd.core.domain.SequenceEvent;
import lombok.AllArgsConstructor;
import org.springframework.stereotype.Component;

@Component
@AllArgsConstructor
public class MessageBuilder implements SequenceEventBuilder {
    private final IdGenerator idGenerator;

    @Override
    public SequenceEvent build(String label, String serviceName, String target, String colour, String data) {
        return new Message(
                idGenerator.next(),
                new ComponentName(serviceName),
                new ComponentName(target),
                label,
                MessageType.SYNCHRONOUS,
                colour,
                data);
    }
}

File: src/main/java/io/lsdconsulting/lsd/distributed/generator/diagram/event/builder/MessageBuilder.kt
package io.lsdconsulting.lsd.distributed.generator.diagram.event.builder

import com.lsd.core.builders.MessageBuilder.Companion.messageBuilder
import com.lsd.core.domain.MessageType
import io.lsdconsulting.lsd.distributed.generator.diagram.event.SequenceEventBuilder

class MessageBuilder : SequenceEventBuilder {
    override fun build(
        id: String,
        label: String,
        serviceName: String,
        target: String,
        colour: String,
        data: String
    ) = messageBuilder()
        .id(id)
        .type(MessageType.SYNCHRONOUS)
        .label(label)
        .from(serviceName)
        .to(target)
        .colour(colour)
        .data(data)
        .build()
}